lxd/storage/backend_lxd.go

```diff
@@ -858,6 +859,15 @@ func (b *lxdBackend) CreateInstanceFromCopy(inst instance.Instance, src instance
 		}
 
 		defer src.Unfreeze()
+
+		// Sync the filesystem.
+		rootfsFile, err := os.Open(src.RootfsPath())
```

Comment:
This pattern of opening a file, syncing and closing has been repeated a fair bit in this PR, would it make sense to move to its own function in `storage/drivers/filesystem/fs.go`?

Reply:
I'm not sure we'd gain a whole lot given the defer would still need to be on the caller side

Reply:
I meant just this bit:

```go
// Sync the filesystem.
rootfsFile, err := os.Open(inst.RootfsPath())
if err != nil {
    return err
}

unix.Syncfs(int(rootfsFile.Fd()))
rootfsFile.Close()
```

Being replaced with something like:

```go
err = filesystem.Syncfs(inst.RootfsPath())
if err != nil {
return err
```

Reply:
Ah yeah, that should be fine.